internal/proxy/fuse_test.go

```diff
@@ -43,21 +43,29 @@ func randTmpDir(t interface {
 // newTestClient is a convenience function for testing that creates a
 // proxy.Client and starts it. The returned cleanup function is also a
 // convenience. Callers may choose to ignore it and manually close the client.
-func newTestClient(t *testing.T, d cloudsql.Dialer, fuseDir, fuseTempDir string) (*proxy.Client, func()) {
+func newTestClient(t *testing.T, d cloudsql.Dialer, fuseDir, fuseTempDir string) (*proxy.Client, chan error, func()) {
 	conf := &proxy.Config{FUSEDir: fuseDir, FUSETempDir: fuseTempDir}
 	c, err := proxy.NewClient(context.Background(), d, testLogger, conf)
 	if err != nil {
 		t.Fatalf("want error = nil, got = %v", err)
 	}
 
 	ready := make(chan struct{})
-	go c.Serve(context.Background(), func() { close(ready) })
+	servErrCh := make(chan error)
+	go func() {
+		servErr := c.Serve(context.Background(), func() { close(ready) })
```

Comment:
My suggestion seems to have broken your test. Maybe it's better to block on the error channel or a context that you can cancel at the end of the test run.

Reply:
There must be some subtle timing difference in the ubuntu implementation of channels. I put the test back to how it was and now it passes. I think its OK for this goroutine to block until `servErrCh` can accept an error.

Reply:
``` golang
ctx, cancel := context.WithCancel(context.Background())
defer cancel()
```

go func() {
  servErr := c.Server(ctx, func() { close(ready) })
  select {
    case servErrCh <- serrErr:
    case <-ctx.Done():
  }
}()
```

Reply:
Fixed.